pkg/agents/team5/agent.go

```diff
@@ -68,22 +70,24 @@ func (a *CustomAgent5) updateSelfishness() {
 	// Tit for tat strategy, agent will conform to the mean behaviour of their social network
 	a.selfishness = 10 - a.calculateAverageFavour()
 	// Make agent less selfish if going through tough times, lowers their expectations and makes them more sympathetic of others struggles
-	a.selfishness = a.restrictToRange(0, 10, a.selfishness-a.daysSinceLastMeal)
+	a.selfishness = restrictToRange(0, 10, a.selfishness-a.daysSinceLastMeal)
 }
 
 func (a *CustomAgent5) updateSatisfaction() {
-	if PercentageHP(a) >= 100 {
-		a.satisfaction = 3
-	}
-	if a.lastMeal == 0 && a.satisfaction > -3 {
-		a.satisfaction--
-	}
-	if PercentageHP(a) < 25 && a.satisfaction > -3 {
-		a.satisfaction--
-	}
-	if PercentageHP(a) > 75 && a.satisfaction < 3 {
-		a.satisfaction++
+	seenFood := a.CurrPlatFood()
+	newSatisfaction := a.satisfaction
+
+	if seenFood >= a.lastSeenFood*12/10 {
```

Comment:
Maybe use 1.2 instead of 12/10 and same for other fractions? Kind of hard to read, feel free to ignore